src/crypto/multi_party_schnorr/traits.rs

```diff
@@ -0,0 +1,116 @@
+use secp256k1::{constants, SecretKey};
+use std::ops::Deref;
+
+pub enum Error {
+    VerifyVSS,
+    VerifyLocalSig,
+}
+
+pub type Message = [u8; 32];
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct Secret([u8; 32]);
+
+impl From<SecretKey> for Secret {
+    fn from(seckey: SecretKey) -> Self {
+        let mut data = [0u8; constants::SECRET_KEY_SIZE];
+        data.copy_from_slice(&seckey[..]);
+        Secret(data)
+    }
+}
+
+/// The schnorr signature what the distributed signing scheme produce finally
+pub struct Signature {
+    /// R.x
+    pub r_x: [u8; 32],
+    /// sigma
+    pub sigma: [u8; 32],
+}
+
+pub mod key_generation_protocol {
+    use crate::crypto::multi_party_schnorr::traits::{Error, Secret};
+    use secp256k1::{PublicKey, SecretKey};
+    use std::collections::HashSet;
+
+    #[derive(Clone, Debug, PartialEq, Eq, Hash)]
+    pub struct NodeVSS {
+        pub sender_pubkey: PublicKey,
+        pub receiver_pubkey: PublicKey,
+        pub commitments: Vec<PublicKey>,
+        pub secret: Secret,
+    }
+
+    /// Struct for a shared secret and an aggregated public key
+    #[derive(Clone, Debug)]
+    pub struct NodeShare {
+        /// Aggregated Public Key
+        pub aggregated_pubkey: PublicKey,
+        /// Secret share for a signer
+        pub secret_share: Secret,
+    }
+
+    pub trait KeyGenerationProtocol {
+        fn create_node_vss(
```

Comment:
I prefer the return type of these functions is Result<...., Error> even if it does not return error now.